chain/substreams/src/mapper.rs

```diff
@@ -1,13 +1,27 @@
-use crate::{Block, Chain, EntityChanges, TriggerData};
+use std::collections::HashMap;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use crate::codec::entity_change;
+use crate::{codec, Block, Chain, EntityChanges, ParsedChanges, TriggerData};
 use graph::blockchain::block_stream::{
     BlockStreamEvent, BlockWithTriggers, FirehoseCursor, SubstreamsError, SubstreamsMapper,
 };
-use graph::prelude::{async_trait, BlockHash, BlockNumber, BlockPtr, Logger};
+use graph::components::store::{EntityKey, EntityType};
+use graph::data::store::scalar::Bytes;
+use graph::data::store::IdType;
+use graph::data::value::Word;
+use graph::data_source::CausalityRegion;
+use graph::prelude::BigDecimal;
+use graph::prelude::{async_trait, BigInt, BlockHash, BlockNumber, BlockPtr, Logger, Value};
+use graph::schema::InputSchema;
 use graph::substreams::Clock;
 use graph::substreams_rpc::response::Message as SubstreamsMessage;
 use prost::Message;
 
-pub struct Mapper {}
+pub struct Mapper {
```

Comment:
It would be good to explain here what this is for, and how setting `schema` to `None` influences things

Reply:
will do!